innerprod/simple/ddh.go

```diff
@@ -172,6 +173,17 @@ func (d *DDH) Decrypt(cipher data.Vector, key *big.Int, y data.Vector) (*big.Int
 		return nil, err
 	}
 
-	return calc.WithBound(bound).BabyStepGiantStep(r, d.Params.g)
+	// TODO: this runs much longer if the result is positive; parallel?
+	res, err := calc.WithBound(bound).BabyStepGiantStep(r, d.Params.g)
```

Comment:
Same comment applies as above.
Perhaps we can have a discussion if implementing a more optimized discrete log algorithm is sensible at this point.